src/wireviz/DataClasses.py

```diff
@@ -214,6 +214,20 @@ def __post_init__(self) -> None:
         else:
             pass  # gauge not specified
 
+        if isinstance(self.length, str):  # length and unit specified
+            try:
+                l, u = self.length.split(' ')
+                l = float(l)
+            except Exception:
+                raise Exception(f'{self.length} - Length must be a number, or number and unit separated by a space')
+            self.length = l
+            self.length_unit = u
+        elif self.length is not None:  # length specified, assume m
+            if self.gauge_unit is None:
+                self.length_unit = 'm'
+        else:
+            pass  # length not specified
```

Comment:
As `length` has `0` as default value, it can only get `None` if specified as `null`, `Null`, `NULL`, or `~` in the input YAML, and that should probably raise an error exception.
```suggestion
        elif not any(isinstance(self.length, t) for t in [int, float]):
            raise Exception(f'Cable {self.name} length has a non-numeric value')
        elif self.gauge_unit is None:
            self.length_unit = 'm'
```
Assigning the default unit is also done below, and is not needed twice.